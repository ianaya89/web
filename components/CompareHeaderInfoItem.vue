<template>
  <li class="compare--dropdown-item">
    <div v-if="$apollo.loading" class="py-4">
      <content-placeholders>
        <content-placeholders-heading :img="true" />
      </content-placeholders>
    </div>
    <compare-item-info
      v-else
      :id="marketplaceItem.id"
      :key="marketplaceItem.id"
      :remover="true"
      :title="$t('app.marketplace.categories.' + $helpers.normalizeCategoryName(marketplaceItem.category.name, marketplaceItem.itemType==='package'))"
      :info="$t('app.marketplace.categories.' + $helpers.normalizeCategoryName($helpers.firstCategory(marketplaceItem.category), marketplaceItem.itemType==='package'))"
    >
      <template v-slot:image>
        <inventory-icon class="w-8 h-8 my-auto" :category="marketplaceItem.category" :is-package="marketplaceItem.itemType==='package'" />
      </template>
    </compare-item-info>
  </li>
</template>

<script>
import { InventoryItemLightQuery } from '@/services/graphql'
import InventoryIcon from '@/components/inventory/InventoryIcon'
import CompareItemInfo from '@/components/CompareItemInfo'

export default {
  name: 'CompareHeaderInfoItem',

  components: {
    CompareItemInfo,
    InventoryIcon
  },

  apollo: {
    marketplaceItem: {
      ...InventoryItemLightQuery,
      variables() {
        return {
          itemId: this.compareItemId
        }
      }
    }
  },

  props: {
    compareItemId: {
      type: Number,
      required: true
    }
  },

  data: () => ({
    marketplaceItem: InventoryItemLightQuery.mock
  })
}
</script>

<style lang="postcss">
.compare--dropdown-item {
  @apply .border-b .border-grey-l4 .px-4;

  &:hover {
    @apply .bg-grey-l4;
  }

  a {
    @apply .flex .items-start .justify-start .px-3 .py-3  w-full .border-l-4 .border-transparent;
    transition: none;
  }

  &-content {
    @apply .flex .flex-col pl-2;
    .name {
      @apply .text-black .text-menu .font-medium .my-0 .py-0;
      line-height: 1.2rem;
    }
    .text {
      @apply .font-serif .text-black .text-p3;
    }
    .time {
      @apply .font-serif .text-p3 .text-grey-l1;
    }
  }
}
</style>
