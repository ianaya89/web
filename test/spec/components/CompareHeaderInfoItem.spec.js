import { shallowMount } from '@vue/test-utils'
import CompareHeaderInfoItem from '@/components/CompareHeaderInfoItem'

describe('CompareHeaderInfoItem', () => {
  it('renders default compare header info', () => {
    const wrapper = shallowMount(CompareHeaderInfoItem, {
      propsData: {
        compareItemId: 10
      }
    })
    expect(wrapper.element).toMatchSnapshot()
  })
})
