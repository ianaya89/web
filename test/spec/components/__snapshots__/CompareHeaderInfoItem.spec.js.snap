// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CompareHeaderInfoItem renders default compare header info 1`] = `
<li
  class="compare--dropdown-item"
>
  <compare-item-info-stub
    info="TRANSLATE:app.marketplace.categories.-"
    remover="true"
    src="\\"thumb-fallback.svg\\""
    title="TRANSLATE:app.marketplace.categories.-"
  />
</li>
`;
